### ISSUE TITLE
Hdlc: Connection Error: Queue overflowed

### ISSUE THREAD

**Issue by keomabrun** [realms-team/solmanager#118]
This appears on Grizzly.
JsonServer from latest smartmeshsdk: https://github.com/twatteyne/smartmeshsdk/commit/3f14c5b9f0feed9e31e94cf693e4fd5c58dcaf76

It seems to be related to https://github.com/realms-team/solmanager/issues/117
```
@Hdlc: Connection Error: Queue overflowed
@Hdlc: Connection Error: Queue overflowed
@Hdlc: Connection Error: Queue overflowed
@Hdlc: Connection Error: Queue overflowed
@Hdlc: Connection Error: Queue overflowed
@Hdlc: Connection Error: Queue overflowed
@Hdlc: Connection Error: Queue overflowed
@Hdlc: Connection Error: Queue overflowed
@Hdlc: Connection Error: Queue overflowed
@Hdlc: Connection Error: Queue overflowed
@Hdlc: Connection Error: Queue overflowed
@Hdlc: Connection Error: Queue overflowed
@Hdlc: Connection Error: Queue overflowed

```

**Comment by keomabrun** [comment]
I think this is related to #119. If we don't see that error on Manager withupdated firmware I will close it.

**Comment by keomabrun** [comment]
```
11 threads running:
- /dev/ttyUSB3_HDLC
- DuplexClientPeriodicTimer
- DustCli
- IpMgrSubscribe
- MainThread
- ManagerHandler@/dev/ttyUSB3
- PubFile
- SnapshotThread
- SolManager
- SolSnapshotThread
- StatsThread
This is thread DustCli.
```

**Comment by keomabrun** [comment]
```
total        used        free      shared  buff/cache   available
Mem:            923          53         556          11         313         804
Swap:            99           0          99

```

**Comment by keomabrun** [comment]
Related reporting: 

* https://dustcloud.atlassian.net/wiki/plugins/servlet/ac/com.atlassian.confluence.plugins.confluence-questions/globalQuestionsViewer?ac.com.atlassian.confluence.plugins.confluence-questions.path=/questions/101220359
* https://dustcloud.atlassian.net/wiki/plugins/servlet/ac/com.atlassian.confluence.plugins.confluence-questions/globalQuestionsViewer?ac.com.atlassian.confluence.plugins.confluence-questions.path=/questions/101220359

**Comment by keomabrun** [comment]
the two lines are always present by pair except a few minutes before the first `Queue overflow` where on the first line is present. Then none of those lines appear again:

```
[urllib3.connectionpool:DEBUG] Starting new HTTP connection (1): api.solsystem.io
2018-07-06 12:51:21,067 [urllib3.connectionpool:DEBUG] http://api.solsystem.io:80 "POST /api/v1/o.json HTTP/1.1" 200 None
```

**Comment by keomabrun** [comment]
There is a time gap before the first `Queue overflow` message. I could not find another similar time gap by manually looking into the previous hours:

```
2018-07-06 12:53:59,094 [solmanager:DEBUG] sending binary object, size: 133 B
2018-07-06 12:54:01,035 [solmanager:DEBUG] sending binary object, size: 33 B
2018-07-06 12:54:03,674 [solmanager:DEBUG] sending binary object, size: 133 B
2018-07-06 12:59:23,286 [Hdlc:ERROR] @Hdlc: Connection Error: Queue overflowed
```

**Comment by keomabrun** [comment]
After setting `period_pollserver_min  = 1` in the config file, the solmanager worked for a month without interruption. Not sure this is linked but I write it here as a reference.

**Comment by keomabrun** [comment]
Should now be fixed by ceb92cd.
The publish call is not threaded and had a blocking queue. An error is now raised if `solmanager` is not fast enough to publish to the `solapi`.
